### DISCUSSION TITLE
Dynamic Network Device Creation

### DISCUSSION THREAD

**Discussion by Ivan-beep** [bpg/terraform-provider-proxmox#1220] [Ideas]
Hello,

I have a use case where a configured number of vlan have to be created.
Those vlans have to be associated to one VM on as much network device.
After checking in the documentation, it is possible to declare various network device statically but nothing using a loop...

The idea is to be able to do something like below : 
```terraform
resource "proxmox_virtual_environment_vm" "kasm_webapp" {
...
  network_device {
    count   = var.num_vlan
    bridge  = var.bridges[count.index]
    vlan_id = var.vlans[count.index]
  }

  initialization {
    datastore_id = "local"

    ip_config {
      ipv4 {
        count   = var.num_vlan
        address = var.ips[count.index]
        gateway = var.gws[count.index]}
      } 
    }
}

```
Do you have any idea how to handle this ? Is it possible to add this feature ?
In the same way, it will be necessary to do the same about `ip_config` section

Thank you

**Comment by bpg-dev** [comment]
Hi @Ivan-beep! 👋🏼

Would something like the config from #798 work for you?

**Comment by Ivan-beep** [comment]
OMG I searched for this but didn't find it... I also forgot to mention I'm a nood on terraform !
Yes that's exactly what I was missing, thank you for your quick answer, much appreciated
Closing the topic now
